src/Microsoft.Health.Fhir.Core/Extensions/RawResourceElementExtensions.cs

```diff
@@ -0,0 +1,108 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.IO;
+using System.Text.Json;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Extensions
+{
+    public static class RawResourceElementExtensions
+    {
+        private static readonly JsonWriterOptions WriterOptions = new JsonWriterOptions { Encoder = System.Text.Encodings.Web.JavaScriptEncoder.UnsafeRelaxedJsonEscaping };
+
+        /// <summary>
+        /// Get the raw data from resourceWrapper as a string and JsonDocument.
+        /// If the RawResource's VersionSet or LastUpdatedSet are false, then the RawResource's data will be updated
+        /// to have them set to the values in the ResourceWrapper
+        /// </summary>
+        /// <param name="rawResource">Input RawResourceElement to convert to a JsonDocument</param>
+        /// <param name="outputStream">Stream to serialize to</param>
+        public static async Task SerializeToStreamAsJson(this RawResourceElement rawResource, Stream outputStream)
+        {
+            EnsureArg.IsNotNull(rawResource, nameof(rawResource));
+
+            if (rawResource.RawResource.IsMetaSet)
+            {
+                using (var sw = new StreamWriter(outputStream, leaveOpen: true))
+                {
+                    await sw.WriteAsync(rawResource.RawResource.Data);
+                    await sw.FlushAsync();
+                    return;
```

Comment:
Instead of explicit calls to FlushAsync before disposing, you can instead do:

``` csharp
                await using (var sw = new StreamWriter(outputStream, leaveOpen: true))
                {
                    await sw.WriteAsync(rawResource.RawResource.Data);
                    return;
                }
```  

which will invoke `DisposeAsync()` instead of `Dispose()`, which in turn calls `FlushAsync()`.

Taking this a step further you can use a "using declaration":
``` csharp
await using var sw = new StreamWriter(outputStream, leaveOpen: true);
await sw.WriteAsync(rawResource.RawResource.Data);
return;
```